src/main/kotlin/kat/auth/AccessManager.kt

```diff
@@ -27,7 +27,10 @@ object AccessManager {
 private fun Context.refreshUserInfo() {
     if (Config.useFakeLogin) return doFakeLogin()
     // if it's not a test-related login we refresh the userinfo (but only if it's already set)
-    this.userInfo?.let { this.userInfo = UserInfo(it.id, AccountService.findById(it.id)!!.role) }
+    this.userInfo?.let {
+        val accountCreate = AccountService.findById(it.id)
+        this.userInfo = accountCreate?.let { UserInfo(this.userInfo!!.id, accountCreate.role) }
```

Comment:
Why are we creating an account here now?

Reply:
The expression "AccountService.findById(it.id)" can return "null". In this case, the "this.userInfo" variable will be set to null. If the expression "AccountService.findById(it.id)" returns a value other than "null", then the "this.userInfo" variable will be assigned a new "UserInfo" object, which will be created using the "accountCreate.role" field.

Reply:
To verify that the application is not working correctly, follow these steps:

Clean up the project. Delete the database file "kotlin-admin-template.db". Launch the application. Open the page http://127.0.0.1:8080/sign-in. Sign up. Stop the application. Delete the database file. Launch the application. Refresh your browser to http://127.0.0.1:8080.

Expected result: page should be displayed: http://127.0.0.1:8080/sign-in

Actual Result: "Internal server error" is displayed.

Reply:
I see, thank you for fixing this :)